### ISSUE TITLE
XmlLicenseRule reads $xml->licence (British spelling) — fails every manifest that follows Joomla's schema, including jedchecker.xml

### ISSUE THREAD

**Issue by uz1mani** [joomla-extensions/jedchecker#278]
## Summary

On `develop` (3.0.0), `XmlLicenseRule` checks `$xml->licence`. Joomla's extension manifest schema uses **`<license>`**. The rule therefore reports `COM_JEDCHECKER_ERROR_XML_LICENCE_NOT_FOUND` for any extension with a correct manifest.

This is a regression: released **2.4.4** reads `$xml->license`.

## Where

`src/administrator/components/com_jedchecker/src/Rule/Rules/XmlLicenseRule.php`, in `find()`:

```php
if (! isset($xml->licence)) {                       // <- c-spelling
    $this->report->addError($file, Text::_('COM_JEDCHECKER_ERROR_XML_LICENCE_NOT_FOUND'));
    return false;
}

if (
    stripos($xml->licence, 'gpl') === false          // <- c-spelling
    && stripos($xml->licence, 'general public licence') === false   // <- c-spelling
) {
```

2.4.4's `libraries/rules/xmllicense.php` had:

```php
if (!isset($xml->license))
...
if (stripos($xml->license, 'gpl') === false
    && stripos($xml->license, 'general public license') === false)
```

So all three occurrences flipped spelling in the refactor.

## Reproduce

Any manifest that follows the schema, e.g. this component's own:

```php
$x = simplexml_load_file('administrator/components/com_jedchecker/jedchecker.xml');
var_dump(isset($x->license));   // true
var_dump(isset($x->licence));   // false  -> rule reports the error
```

`jedchecker.xml` declares `<license>GNU General Public License version 2 or later; see LICENSE.txt</license>`, so **JED Checker fails its own manifest**. Joomla core's `administrator/manifests/files/joomla.xml` and `com_content/content.xml` also use `<license>`.

Adding a `<licence>` element to an otherwise unchanged extension takes the run from 1 error to 0.

## Second, separate effect

Even with the element found, the GPL match looks for `'general public licence'`. The conventional Joomla string is `GNU General Public License version 2 or later`, which contains neither `gpl` nor `general public licence`, so it then falls through to `addCompat(COM_JEDCHECKER_ERROR_XML_LICENCE_NOT_GPL)`.

## Suggested fix

Accept both spellings rather than swapping back, since `<licence>` may exist in the wild:

```php
$license = $xml->licence ?? $xml->license ?? null;
if ($license === null) { /* error */ }
$l = (string) $license;
if (stripos($l, 'gpl') === false
    && stripos($l, 'general public license') === false
    && stripos($l, 'general public licence') === false) { /* compat */ }
```

## Environment

Joomla 6.1.3, PHP 8.5.10, jedchecker `develop` built with `vendor/bin/robo build` (reports itself as 3.0.0).

## Unrelated, noticed while testing

The built package's view requests `media/com_jedchecker/js/jedchecker.js`, but the build ships `media/com_jedchecker/js/script.js`. The 404 leaves the upload button unwired, so the results panel renders its default all-passed state and never runs. Happy to open that separately if useful.
